### DISCUSSION TITLE
Merge imports

### DISCUSSION THREAD

**Discussion by AndreiTS** [biomejs/biome#454] [General]
For example:
It would be nice if biome could merge imports from the same place. Example:

``` 
import { A } from 'package';
import { B } from 'package';
```

would become:

`import { A, B } from 'package';`

**Comment by ematipico** [comment]
Do you know of any prior work? Eslint plugin, prettier, clippy, etc.?

**Reply by Conaclos** [reply]
`Organize imports` action from the TypeScript LSP merges imports from the same source. The imports must be in the same import group. I don't know for other tools.

**Reply by AndreiTS** [reply]
before biome, I used prettier/eslint with typescript, and when I saved a file it would merge imports. Idk what package was doing what.

**Reply by AndreiTS** [reply]
I think rust fmt doest that: https://github.com/rust-lang/rustfmt/issues/1383

**Reply by victor-teles** [reply]
I was using eslint-plugin-import to do that
https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/no-duplicates.md

**Comment by Conaclos** [comment]
@arendjr Any thoughts?

**Reply by arendjr** [reply]
I agree this is a good thing to have :+1: I suppose it makes sense to limit it to the same import group indeed.

**Comment by william-ford-rq** [comment]
Any progress on this? I would love for this to be a Biome feature :)
Right now I'm doing this by searching for duplicate import sources using regex `/import .* from '(.*)';\nimport .* from '\1';/gm`

**Reply by Conaclos** [reply]
I have started writing an RFC to improve/revise our organise import feature. This will be part of it.

**Comment by sterliakov** [comment]
Was this missed from 2.0.0-beta.1 implementation? I just tried to upgrade but still have to keep eslint `import` plugin with [this only rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md) enabled:(

```
import {a} from 'foo';
import {b} from 'foo';
```

remains two separate lines

**Reply by Conaclos** [reply]
Yes it is not yet implemented.

**Comment by Conaclos** [comment]
I [opened a PR](https://github.com/biomejs/biome/pull/5579) to handle import and export merging.
